android/WALT/app/src/main/java/org/chromium/latency/walt/AutoRunFragment.java

```diff
@@ -28,14 +28,56 @@
 import android.view.ViewGroup;
 import android.widget.TextView;
 
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.ArrayList;
+
 public class AutoRunFragment extends Fragment {
 
     static final String TEST_ACTION = "org.chromium.latency.walt.START_TEST";
 
     private ClockManager clockManager;
     private AudioTest toTearDown; // TODO: figure out a better way to destroy the engine
 
+    private class AudioResultHandler implements ResultHandler {
+        private FileWriter fileWriter;
+
+        AudioResultHandler(String fileName) throws IOException {
+            fileWriter = new FileWriter(fileName);
+        }
+
+        @Override
+        public void onResult(ArrayList<Double> r) {
+            logger.log("Writing data file");
+            try {
+                for (double v : r) {
+                    fileWriter.write(v + ",\n");
```

Comment:
Looks like it's a CSV with a single column. I think in this case the comma is not necessary.